source/DistanceMap.h

```diff
@@ -24,44 +25,48 @@ class System;
 
 
 
-// This is a map of how many hyperspace jumps it takes to get to other systems
-// from the given "center" system. Ships with a hyperdrive travel using the
-// "links" between systems. Ships with jump drives can make use of those links,
-// but can also travel to any of a system's "neighbors." A distance map can also
-// be used to calculate the shortest route between two systems.
+// This is a map of shortest routes to get to other systems from the given "center"
+// system. It also tracks how many days and fuel it takes to get there. Ships with
+// a hyperdrive travel using the "links" between systems. Ships with jump drives
+// can make use of those links, but can also travel to any system that's closeby.
+// Wormholes can also be used by players or ships.
 class DistanceMap {
 public:
-	// Find paths to the given system. The optional arguments put a limit on how
-	// many systems will be returned and how far away they are allowed to be.
-	explicit DistanceMap(const System *center, int maxCount = -1, int maxDistance = -1);
+	// Find paths branching out from the given system. The optional arguments put
+	// a limit on how many systems will be returned (e.g. buying a local map)
+	// and how far away they are allowed to be (e.g. a valid mission location)
+	explicit DistanceMap(const System *center, int maxSystems = -1, int maxDays = -1);
 	// If a player is given, the map will only use hyperspace paths known to the
 	// player; that is, one end of the path has been visited. Also, if the
 	// player's flagship has a jump drive, the jumps will be make use of it.
-	explicit DistanceMap(const PlayerInfo &player, const System *center = nullptr);
+	// Optionally, the pathfinding will stop once a path to the destination is found.
+	explicit DistanceMap(const PlayerInfo &player, const System *center = nullptr, const System *destination = nullptr);
 	// Calculate the path for the given ship to get to the given system. The
-	// ship will use a jump drive or hyperdrive depending on what it has. The
-	// pathfinding will stop once a path to the destination is found.
+	// ship will use a jump drive or hyperdrive depending on what it has.
 	DistanceMap(const Ship &ship, const System *destination);
 
-	// Find out if the given system is reachable.
-	bool HasRoute(const System *system) const;
+	// Find out if the given system is reachable, defaulting to 'destination'
+	bool HasRoute(const System *system = nullptr) const;
```

Comment:
We should try to avoid pointers where it doesn't really make sense to use them. HasRoute should take the system by reference and you can introduce a new overload without parameters that uses destination. Same thing for the other functions below.

Also since not every DistanceMap has a destination specified, maybe consider not prodiving a way to call the function which uses destination by default to avoid confusion if used accidentally on a DistanceMap without destination.

Reply:
I imagine "not prodiving a way to call the function which uses destination" would require a subclass. That's doable.